### ISSUE TITLE
Frame vs Series start time

### ISSUE THREAD

**Issue by philrelton** [gwpy/gwpy#1431]
Hi, I've been having some problems with writing to file once changing the start time of a TimeSeries. I belive that the problem is that the series time is being changed and the frame is not changing with it. I have made a small working example of this below:
```python
import numpy as np
from gwpy.timeseries import TimeSeries
from astropy.units.quantity import Quantity

# Setting up some fake data
data = TimeSeries(np.zeros(10*16384))
data.t0 = 1260000000
data.dt = 1 / 16384

# Applying the start time conversion to match the required time samples
actual_start = Quantity(1260000013.559644,"s")
offset = (data.t0 - actual_start) % data.dt
data.t0 += data.dt - offset

# Attempting to write the data
data.write("test.gwf")
```
However, upon doing this I get the following error:
```
XLAL Error - XLALFrameAddREAL8TimeSeriesProcData (/home/conda/feedstock_root/build_artifacts/lalframe-split_1613814628945/work/lib/LALFrameIO.c:601): Series start time 1260000000.000012600 is earlier than frame start time 1260000000.000012636
XLAL Error - XLALFrameAddREAL8TimeSeriesProcData (/home/conda/feedstock_root/build_artifacts/lalframe-split_1613814628945/work/lib/LALFrameIO.c:601): Invalid argument

---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
/local/philip.relton/ipykernel_96008/2839776775.py in <module>
      1 # Attempting to write the data
----> 2 data.write("test.gwf")

~/.conda/envs/pipelines/lib/python3.7/site-packages/gwpy/timeseries/core.py in write(self, target, *args, **kwargs)
    325         Notes
    326         -----"""
--> 327         return io_registry.write(self, target, *args, **kwargs)
    328 
    329     @classmethod

~/.conda/envs/pipelines/lib/python3.7/site-packages/astropy/io/registry.py in write(data, format, *args, **kwargs)
    568 
    569     writer = get_writer(format, data.__class__)
--> 570     writer(data, *args, **kwargs)
    571 
    572 

~/.conda/envs/pipelines/lib/python3.7/site-packages/gwpy/timeseries/io/gwf/__init__.py in write_(*args, **kwargs)
    408         fmt = 'gwf.{}'.format(get_default_gwf_api())
    409         writer = get_writer(fmt, container)
--> 410         return writer(*args, **kwargs)
    411 
    412     register_identifier('gwf', container, identify_gwf)

~/.conda/envs/pipelines/lib/python3.7/site-packages/gwpy/timeseries/io/gwf/__init__.py in write_timeseries(series, *args, **kwargs)
    373             for available arguments and keyword arguments
    374         """
--> 375         return write_timeseriesdict({None: series}, *args, **kwargs)
    376 
    377     # -- register -------------------------------

~/.conda/envs/pipelines/lib/python3.7/site-packages/gwpy/timeseries/io/gwf/__init__.py in write_timeseriesdict(data, outfile, start, end, type, **kwargs)
    362             end=end,
    363             type=type,
--> 364             **kwargs,
    365         )
    366 

~/.conda/envs/pipelines/lib/python3.7/site-packages/gwpy/timeseries/io/gwf/lalframe.py in write(tsdict, outfile, start, end, type, name, run)
    223 
    224         # add time series to frame
--> 225         add_(frame, lalseries)
    226 
    227     # write frame

RuntimeError: Invalid argument
```

<details>
<summary>FrameCPP install error</summary>

By the suggestion of @duncanmmacleod, I installed `python-ldas-tools-framecpp`. This provided me with the more confusing error:
```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
/local/philip.relton/ipykernel_86753/528858322.py in <module>
----> 1 data.write("test.gwf")

~/.conda/envs/pipelines/lib/python3.7/site-packages/gwpy/timeseries/core.py in write(self, target, *args, **kwargs)
    325         Notes
    326         -----"""
--> 327         return io_registry.write(self, target, *args, **kwargs)
    328 
    329     @classmethod

~/.conda/envs/pipelines/lib/python3.7/site-packages/astropy/io/registry.py in write(data, format, *args, **kwargs)
    568 
    569     writer = get_writer(format, data.__class__)
--> 570     writer(data, *args, **kwargs)
    571 
    572 

~/.conda/envs/pipelines/lib/python3.7/site-packages/gwpy/timeseries/io/gwf/__init__.py in write_(*args, **kwargs)
    408         fmt = 'gwf.{}'.format(get_default_gwf_api())
    409         writer = get_writer(fmt, container)
--> 410         return writer(*args, **kwargs)
    411 
    412     register_identifier('gwf', container, identify_gwf)

~/.conda/envs/pipelines/lib/python3.7/site-packages/gwpy/timeseries/io/gwf/__init__.py in write_timeseries(series, *args, **kwargs)
    373             for available arguments and keyword arguments
    374         """
--> 375         return write_timeseriesdict({None: series}, *args, **kwargs)
    376 
    377     # -- register -------------------------------

~/.conda/envs/pipelines/lib/python3.7/site-packages/gwpy/timeseries/io/gwf/__init__.py in write_timeseriesdict(data, outfile, start, end, type, **kwargs)
    362             end=end,
    363             type=type,
--> 364             **kwargs,
    365         )
    366 

~/.conda/envs/pipelines/lib/python3.7/site-packages/gwpy/timeseries/io/gwf/framecpp.py in write(tsdict, outfile, start, end, type, name, run, compression, compression_level)
    523         else:
    524             kw = {}
--> 525         _append_to_frame(frame, tsdict[key].crop(start, end), type=ctype, **kw)
    526 
    527     # write frame to file

~/.conda/envs/pipelines/lib/python3.7/site-packages/gwpy/timeseries/io/gwf/framecpp.py in _append_to_frame(frame, timeseries, type, **kwargs)
    576         raise RuntimeError("Invalid channel type {!r}, please select one of "
    577                            "'adc, 'proc', or 'sim'".format(type))
--> 578     frdata = create(timeseries, frame_epoch=epoch, **kwargs)
    579 
    580     # append an FrVect

~/.conda/envs/pipelines/lib/python3.7/site-packages/gwpy/io/gwf.py in create_frprocdata(series, frame_epoch, comment, type, subtype, trange, fshift, phase, frange, bandwidth)
    328         phase,
    329         frange,
--> 330         bandwidth,
    331     )
    332 

~/.conda/envs/pipelines/lib/python3.7/site-packages/LDAStools/frameCPP.py in __init__(self, *args)
   1524         1
   1525         """
-> 1526         _frameCPP.FrProcData_swiginit(self, _frameCPP.new_FrProcData(*args))
   1527     GetName = _swig_new_instance_method(_frameCPP.FrProcData_GetName)
   1528     GetComment = _swig_new_instance_method(_frameCPP.FrProcData_GetComment)

TypeError: in method 'new_FrProcData', argument 7 of type 'FrProcData::fShift_type'
Additional information:
Wrong number or type of arguments for overloaded function 'new_FrProcData'.
  Possible C/C++ prototypes are:
    FrProcData::FrProcData()
    FrProcData::FrProcData(std::string const &,std::string const &,FrProcData::type_type,FrProcData::subType_type,FrProcData::timeOffset_type,FrProcData::tRange_type,FrProcData::fShift_type,FrProcData::phase_type,FrProcData::fRange_type,FrProcData::BW_type)
```

Package versions were:
Name                    Version                   Build  Channel
numpy                     1.21.1           py37h038b26d_0    conda-forge
gwpy                      2.0.4              pyhd8ed1ab_1    conda-forge
astropy                   4.3.1            py37hb1e94ed_1    conda-forge
astropy-healpix           0.6              py37hb1e94ed_0    conda-forge

I uninstalled `python-ldas-tools-framecpp` in order to produce this, but I belived the version was `2.6.12`

</details>

**Comment by duncanmmacleod** [comment]
I can reproduce this bug when specifically choosing the `gwf.lalframe` GWF format (which is the default iff you have `python-lalframe` installed but **not** `python-ldas-tools-framecpp`).

**Comment by duncanmmacleod** [comment]
This has been reported upstream as https://git.ligo.org/lscsoft/lalsuite/-/issues/488.

**Comment by duncanmmacleod** [comment]
This should have been resolved with the LALFrame 2.0.0.

**Comment by duncanmmacleod** [comment]
Migrated to https://gitlab.com/gwpy/gwpy/-/issues/1431, please join us on gitlab to continue your contribution!
